### PR TITLE
Fix geovista integration test

### PR DESCRIPTION
**Pull request by bjlittle**
### Overview

This pull-request updates how the `geovista` integration test is executed after [geovista PR#1494](https://github.com/bjlittle/geovista/pull/1494) finally introduced the minimum pin `pyvista>=0.45.0` and changed how the headless display is configured.

Essentially `geovista` dropped the use of the `pytest` plugin [pytest-xvfb](https://github.com/The-Compiler/pytest-xvfb) and adopted the pattern of using the `pyvista/setup-headless-display-action` GHA and `xvfb-run` combination, akin to `pyvista`.

Once this is merged, blocked PRs can rebase from `main` to pickup this fix :+1:

### CODE REVIEW

**Comment by bjlittle** [comment]
@user27182 Apologies for the blip :blush: 

Thanks for the nudge :beers:

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -104,11 +104,7 @@ jobs:
       CARTOPY_SHARE_DIR: ~/.local/share/cartopy
       GEOVISTA_POOCH_MUTE: true
     steps:
-      - uses: awalsh128/cache-apt-pkgs-action@v1.5.0
-        if: env.USE_CACHE == 'true'
-        with:
-          packages: xvfb
-          version: 3.0
+      - uses: pyvista/setup-headless-display-action@main
```
Comment (bjlittle) [inline .github/workflows/integration-tests.yml:107]: Do you guys do any kind of caching here? Or always roll with the latest `apt update`  packages from within the `setup-headless-display-action` every time?

File: .github/workflows/integration-tests.yml [hunk 2]
```diff
@@ -124,8 +120,8 @@ jobs:
       - name: Download cartopy assets
         run: |
           mkdir -p ${CARTOPY_SHARE_DIR}
-          python -m cartopy.feature.download physical --output ${CARTOPY_SHARE_DIR} --no-warn
-      - run: pytest
+          cartopy_feature_download physical --output ${CARTOPY_SHARE_DIR} --no-warn
```
Comment (bjlittle) [inline .github/workflows/integration-tests.yml:123]: There is now an entry point available in `cartopy` which is a tad neater :wink:

**Comment by bjlittle** [comment]
Dang, the `pyvista` docs are an expensive beast to build :fearful: 

Are there any thoughts on allowing the ability to skip building the docs?

It might expedite corner case situations where it's safe and appropriate to do that ... :thinking:

File: .github/workflows/integration-tests.yml (hunk 1 above)
Comment (user27182) [reply .github/workflows/integration-tests.yml:107]: no cache is fine here, `env.USE_CACHE` is used to protect from cache poisoning when caching is used

**Review by user27182 — APPROVED** [review]
Thanks @bjlittle

File: .github/workflows/integration-tests.yml (hunk 1 above)
Comment (bjlittle) [reply .github/workflows/integration-tests.yml:107]: @user27182 Just for my own education ... what causes cache poisoning?

Comment (user27182) [reply .github/workflows/integration-tests.yml:107]: See https://docs.zizmor.sh/audits/#cache-poisoning
We added `zizmor` in https://github.com/pyvista/pyvista/pull/7006, I recommend reading [Supply-chain attack analysis: Ultralytics](https://blog.pypi.org/posts/2024-12-11-ultralytics-attack-analysis/) for motivation behind enabling this check

**Comment by bjlittle** [comment]
@user27182 Over to you 👍

**Review by MatthewFlamm — APPROVED** [review]

**Comment by bjlittle** [comment]
Awesome, thanks @user27182 and @MatthewFlamm 🍻
